msg/sender-not-used-in-transferfrom.yaml

```diff
@@ -0,0 +1,16 @@
+rules:
+- id: msg.sender-not-used-in-transferFrom
+  patterns:
+  - pattern-regex: transferFrom\(
+  - pattern-not-regex: transferFrom\(msg.sender
+  message: msg.sender is not being used when calling erc20.transferFrom
+  languages:
+  - solidity
+  severity: WARNING
+  metadata:
+    category: security
+    technology:
+    - solidity
+    cwe: 'CWE 285: The software does not perform or incorrectly performs an authorization
```

Comment:
We just need the CWE phrase `[CWE-285: Improper Authorization](https://cwe.mitre.org/data/definitions/285.html)`